client/components/course/Sidebar/Relationship.vue

```diff
@@ -67,11 +71,15 @@ export default {
       }
       return filter(activities, it => every(conds, cond => cond(it)));
     },
-    optionGroups() {
-      return map(groupBy(this.options, 'type'), (it, type) => ({
-        typeLabel: getLevel(type).label,
-        activities: it
-      }));
+    groupedOptions() {
+      const grouped = groupBy(this.options, 'type');
+      const withTypes = map(grouped, (it, type) => {
```

Comment:
`reduce` or `flatMap` can be used here instead of a `map` plus custom `flatten` method.